test/debug_info/parameter.f90

```diff
@@ -0,0 +1,17 @@
+!RUN: %flang -g -S -emit-llvm %s -o - | FileCheck %s
+
+!CHECK: call void @llvm.dbg.value(metadata i64 99, metadata [[SPAR:![0-9]+]], metadata !DIExpression())
+!CHECK: distinct !DIGlobalVariable(name: "apar"
```

Comment:
I am not sure what I am missing here. But for me it shows the expected result regardless of dwarf version.

 ----------------------------------
# flang parameter.f90 -gdwarf-2 -S -emit-llvm
# grep -e "spar" -e "apar" -e "Dwarf Version" parameter.ll
!8 = distinct !DIGlobalVariable(name: "apar", scope: !2, file: !3, type: !9, isLocal: true, isDefinition: true)
!15 = !{i32 2, !"Dwarf Version", i32 2}
!17 = !DILocalVariable(name: "spar", scope: !18, file: !3, type: !10)
  ------------------------------------
I am using this driver.

$ git branch
  70_mod
  master
  release_60
* release_70
$ git remote -v
origin  https://github.com/flang-compiler/flang-driver.git (fetch)
origin  https://github.com/flang-compiler/flang-driver.git (push)
----------------- 
Can you please share the details or your sources (open) ?
In case it works for you on dwarf-4 (with DWARF related PRs), it would be better to first commit below PRs
#863 #89 #90 #91

Reply:
There are two separate issues here.
1) Not able to see !8 = distinct !DIGlobalVariable(name: "apar", scope: !2, file: !3, type: !9, isLocal: true, isDefinition: true)
This is because i am building with FLANG_LLVM_EXTENSIONS. If i do not use that cmake variable then I can see the debug for apar.

2) Not able to see values of spar in gdb with default dwarf version =2.
```
Breakpoint 1, main () at param.f90:6
6         svar = spar
(gdb) info locals
svar = 0
spar = <error reading variable spar (Cannot access memory at address 0x63)>
apar = (99, 98, 97, 96, 95)
avar = (0, 0, 0, 0, 0)
```
If i manually change the dwarf version to 4 in the LLVM IR generated by flang then I get the following in gdb.
```
Breakpoint 1, main () at param.f90:6
6         svar = spar
(gdb) info locals
svar = 0
spar = 99
apar = (99, 98, 97, 96, 95)
avar = (0, 0, 0, 0, 0)
```

Reply:
Thanks, this helps. 
Regarding first issue, I never tested with FLANG_LLVM_EXTENSION. Now I am able to reproduce this issue. I shall shortly push the fix to make it work with the FLANG_LLVM_EXTENSION as well.